optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java

```diff
@@ -33,14 +45,195 @@
+                                        Files.createDirectories(classesDir);
+                                        return classesDir;
+                                    } catch (IOException e) {
+                                        // ignore and return null
+                                    }
+                                }
+                                return null;
+                            }
+
+                            private boolean isFoundInLocalRepo(Artifact artifact) {
+                                String localRepo;
+                                try {
+                                    localRepo = mvnCtx.getLocalRepo();
+                                } catch (BootstrapMavenException e) {
+                                    throw new IllegalStateException(e);
+                                }
+                                if (localRepo == null) {
+                                    return false;
+                                }
+                                Path p = Paths.get(localRepo);
+                                for (String s : artifact.getGroupId().split("\\.")) {
+                                    p = p.resolve(s);
+                                }
+                                p = p.resolve(artifact.getArtifactId());
+                                p = p.resolve(artifact.getVersion());
+                                p = p.resolve(LocalWorkspace.getFileName(artifact));
+                                return Files.exists(p);
+                            }
+
+                            @Override
+                            public List<String> findVersions(Artifact artifact) {
+                                return ws.findVersions(artifact);
+                            }
+                        });
+
+                        final MavenArtifactResolver resolver = MavenArtifactResolver.builder()
+                                .setRepositorySystem(mvnCtx.getRepositorySystem())
+                                .setRepositorySystemSession(session)
+                                .setRemoteRepositories(mvnCtx.getRemoteRepositories())
+                                .setRemoteRepositoryManager(mvnCtx.getRemoteRepositoryManager())
+                                .setCurrentProject(currentProject)
+                                .build();
+                        System.out
+                                .println("  " + resolver
+                                        .resolve(new DefaultArtifact(currentProject.getGroupId(),
+                                                currentProject.getArtifactId(), "jar", currentProject.getVersion()))
+                                        .getArtifact().getFile());
+                    } catch (BootstrapMavenException e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                }
+            })
```

Comment:
Do we really need all this code? Less is more.
Do we really need to parse the maven metadata and start manipulating files?